### PR TITLE
Leverage class name literal on object

### PR DESCRIPTION
**Pull request by tigitz**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       |  
| License       | MIT
| Doc PR        |

### Rational
[PHP RFC: Allow ::class on objects](https://wiki.php.net/rfc/class_name_literal_on_object)

More specifically
>  As it is syntactically similar to a class constant access, programmers intuitively expect the syntax $object::class to work as well and provide the same result as get_class($object). This RFC proposes to allow that syntax.

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Tests/DependencyInjection/DoctrineExtensionTest.php [hunk 1]
```diff
@@ -77,7 +77,7 @@ public function testFixManagersAutoMappingsWithTwoAutomappings()
             'SecondBundle' => 'My\SecondBundle',
         ];
 
-        $reflection = new \ReflectionClass(\get_class($this->extension));
+        $reflection = new \ReflectionClass($this->extension::class);
```
Comment (Koc) [inline src/Symfony/Bridge/Doctrine/Tests/DependencyInjection/DoctrineExtensionTest.php]: actually `ReflectionClass` constructor supports objects as well

Comment (nicolas-grekas) [reply src/Symfony/Bridge/Doctrine/Tests/DependencyInjection/DoctrineExtensionTest.php]: Right! Let use the object directly everywhere possible.

Comment (tigitz) [reply src/Symfony/Bridge/Doctrine/Tests/DependencyInjection/DoctrineExtensionTest.php]: Good catch , thanks

**Comment by tigitz** [comment]
@Koc @nicolas-grekas Updated and rebased on 6.3, test failure seems unrelated, ready to be reviewed/merged. Thanks :pray:

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @tigitz.

**Pull request merged**